### PR TITLE
Add support for creating folders and creating dashboards inside folders

### PR DESCRIPTION
**Pull request by goraxe**

### CODE REVIEW

**Comment by goraxe** [comment]
cc @tonglil, @msuterski, @sbower @mlclmj

**Comment by tonglil** [comment]
Sorry @goraxe, just saw this!

File: grafana/provider.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 
-	gapi "github.com/nytm/go-grafana-api"
+	gapi "github.com/goraxe/go-grafana-api"
```
Comment (tonglil) [inline grafana/provider.go]: Just merged the changes to the library.

**Comment by mlclmj** [comment]
Hi all! Sorry for the delay looking at this, I'll take a look at this this evening and leave any comments I have

File: grafana/resource_dashboard.go [hunk 2]
```diff
@@ -22,6 +22,14 @@ func ResourceDashboard() *schema.Resource {
 				Computed: true,
 			},
 
+			"folder": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Required: false,
+				ForceNew: true,
+				Computed: false,
```
Comment (mlclmj) [inline grafana/resource_dashboard.go]: My understanding has been that it's generally only necessary to include fields which need to be set to `true`, was there a reason for explicitly including `Required` and `Computed`?

File: grafana/resource_folder.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"uid": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Required: false,
```
Comment (mlclmj) [inline grafana/resource_folder.go]: Similar to my previous comment, do we need `Required: false` here?

File: grafana/resource_dashboard_test.go [hunk 4]
```diff
@@ -75,6 +102,16 @@ func testAccDashboardCheckExists(rn string, dashboard *gapi.Dashboard) resource.
 	}
 }
 
+func testAccDashboardCheckExistsInFolder(dashboard *gapi.Dashboard, folder *gapi.Folder) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// return fmt.Errorf("folder: %#v dashboard: %#v", folder, dashboard)
```
Comment (mlclmj) [inline grafana/resource_dashboard_test.go]: Is this line needed?

**Review by mlclmj — COMMENTED** [review]
@goraxe Thanks for submitting this PR! A couple little things, but other than that this is looking good to me.

File: grafana/resource_dashboard.go [hunk 5]
```diff
@@ -7,7 +7,7 @@ import (
 
 	"github.com/hashicorp/terraform/helper/schema"
 
-	gapi "github.com/nytm/go-grafana-api"
+	gapi "github.com/goraxe/go-grafana-api"
```
Comment (mlclmj) [inline grafana/resource_dashboard.go]: Looks like this one slipped back in

**Comment by goraxe** [comment]
Hi @tonglil @mlclmj any more review comments?   Thanks :-)

**Comment by tonglil** [comment]
Thanks for your contribution!

**Pull request merged**